metrics/bindinfo.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// bindinfo metrics.
+var (
+	BindUsageCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "bindinfo",
+			Name:      "bind_usage_counter",
+			Help:      "Counter of query using sql bind",
+		}, []string{LblType})
+
+	BindTotalGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "tidb",
+			Subsystem: "bindinfo",
+			Name:      "bind_total_gauge",
+			Help:      "Total number of sql bind",
+		}, []string{LblType})
```

Comment:
Can this value be retrieved directly from system tables?

Reply:
Yes, it can, but the system table only contains global level bindings.